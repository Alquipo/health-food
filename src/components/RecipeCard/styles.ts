import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: 52.8rem;
    height: 22.5rem;
    width: 100%;
    display: flex;
    align-items: center;
    border-radius: ${theme.border.radius};
    overflow: hidden;
    box-shadow: 0 1.1rem 1.5rem 0 rgba(158, 154, 154, 0.36);

    ${media.lessThan('small')`
      max-width: 100%;
      flex-direction: column;
      height: 40rem;

    `};
  `}
`

export const WrapperImage = styled.div`
  flex: 9;
  width: 100%;
  height: 100%;
  position: relative;

  ${media.lessThan('small')`
    flex: 9;
  `};
`

export const Content = styled.div`
  ${({ theme }) => css`
    flex: 8;
    padding: 0 ${theme.spacings.small};

    ${media.lessThan('small')`
      flex: 5;
      display: flex;
      flex-direction: column;
      align-items: center;
      text-align: center;
      padding: ${theme.spacings.xsmall};
      margin-top: ${theme.spacings.xsmall};
    `};
  `}
`

export const Title = styled.h3`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xsmall};
    color: ${theme.colors.secondary};
    letter-spacing: 0.1rem;
  `}
`
