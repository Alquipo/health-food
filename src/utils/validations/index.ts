import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
