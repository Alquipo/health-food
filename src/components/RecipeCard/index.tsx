import * as S from './styles'

import Button from '../Button'
import Image from 'next/image'

export type RecipeCardProps = {
  image: string
  title: string
}

export default function RecipeCard({ image, title }: RecipeCardProps) {
  return (
    <S.Wrapper>
      <S.WrapperImage>
        <Image src={image} alt={title} objectFit="cover" layout="fill" />
      </S.WrapperImage>

      <S.Content>
        <S.Title>{title}</S.Title>
        <Button>See Recipe</Button>
      </S.Content>
    </S.Wrapper>
  )
}
