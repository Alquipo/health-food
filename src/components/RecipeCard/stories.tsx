import { Story, Meta } from '@storybook/react'
import RecipeCard, { RecipeCardProps } from '.'

export default {
  title: 'RecipeCard',
  component: RecipeCard
} as Meta

export const Default: Story<RecipeCardProps> = (args) => (
  <RecipeCard {...args} />
)

Default.args = {
  image: '/img/comida_1.svg',
  title: 'Broccoli Saladwith Bacon'
}
