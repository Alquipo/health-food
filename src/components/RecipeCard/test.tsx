import { render, screen } from 'utils/test-utils'

import RecipeCard from '.'

const props = {
  image: '/img/comida_1.svg',
  title: 'Broccoli Saladwith Bacon'
}

describe('<RecipeCard/>', () => {
  it('should render a service section correctly', () => {
    const { container } = render(<RecipeCard {...props} />)

    expect(
      screen.getByRole('img', {
        name: props.title
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: props.title
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('button', { name: /See Recipe/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
