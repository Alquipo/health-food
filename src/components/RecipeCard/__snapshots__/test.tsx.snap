// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeCard/> should render a service section correctly 1`] = `
.c0 {
  max-width: 52.8rem;
  height: 22.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.6rem;
  overflow: hidden;
  box-shadow: 0 1.1rem 1.5rem 0 rgba(158,154,154,0.36);
}

.c1 {
  -webkit-flex: 9;
  -ms-flex: 9;
  flex: 9;
  width: 100%;
  height: 100%;
  position: relative;
}

.c2 {
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  padding: 0 2.4rem;
}

.c3 {
  margin-bottom: 1.6rem;
  color: #1D164D;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c4 {
  padding: 1.6rem 2.4rem;
  background: #BADC58;
  color: #FFFFFF;
  border-radius: 0.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  cursor: pointer;
  border: none;
  font-size: 1.8rem;
  font-weight: 600;
}

.c4:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

@media (max-width:450px) {
  .c0 {
    max-width: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: 40rem;
  }
}

@media (max-width:450px) {
  .c1 {
    -webkit-flex: 9;
    -ms-flex: 9;
    flex: 9;
  }
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex: 5;
    -ms-flex: 5;
    flex: 5;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    text-align: center;
    padding: 1.6rem;
    margin-top: 1.6rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <noscript />
      <img
        alt="Broccoli Saladwith Bacon"
        data-nimg="true"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
    </div>
  </div>
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      Broccoli Saladwith Bacon
    </h3>
    <button
      class="c4"
      type="button"
    >
      See Recipe
    </button>
  </div>
</div>
`;
